src/charm.py

```diff
@@ -410,6 +411,10 @@ def _on_database_created(self, event: DatabaseCreatedEvent) -> None:
             self.unit.status = WaitingStatus("Waiting to connect to Hydra container")
             return
 
+        if not self._validate_config_log_level():
+            event.defer()
+            return
+
```

Comment:
could you add a unit test for it?

Reply:
I took this out, for log_level the only relevant event handlers are pebble_ready and config_change.